tests/php/modules/geo-location/test_class.jetpack-geo-location.php

```diff
@@ -0,0 +1,365 @@
+	}
+
+	public function test_the_content_location_display_escapes_malicious_meta_values() {
+		$instance = $this->get_instance_with_mock_malicious_post();
+
+		$instance->method( 'is_single' )->willReturn( true );
+
+		$modifiedContent = $instance->the_content_location_display( 'Original content' );
+
+		$this->assertStringStartsWith( 'Original content', $modifiedContent);
+		$this->assertNotContains( '<attack>', $modifiedContent );
+		$this->assertContains( '&lt;', $modifiedContent );
+		$this->assertContains( '&gt;', $modifiedContent );
+	}
+
+	private function get_instance() {
+		return Jetpack_Geo_Location::init();
+	}
+
+	private function get_instance_with_mock_public_post() {
+		$instance = $this->create_mock_instance();
+
+		$instance->method( 'get_meta_value' )
+	         ->will(
+	         	$this->returnValueMap(
+	         	    array(
+	         	    	array( 1, 'public', '1' ),
+		                array( 1, 'latitude', self::MOCK_LAT ),
+		                array( 1, 'longitude',  self::MOCK_LONG ),
+		                array( 1, 'address', self::MOCK_ADDRESS )
+	                )
+	            )
+	         );
+
+		return $instance;
+	}
+
+	private function get_instance_with_mock_malicious_post() {
+		$instance = $this->create_mock_instance( array( 'get_meta_values' ) );
+
+		$instance->method( 'get_meta_values' )
+	         ->will(
+		         $this->returnValue(
+			         array(
+				         'is_public'    => true,
+				         'latitude'     => '<attack>',
+				         'longitude'    => '<attack>',
+				         'label'        => '<attack>',
+				         'is_populated' => true,
+			         )
+		         )
+	         );
+
+		return $instance;
+	}
+
+	private function get_instance_with_mock_private_post() {
+		$instance = $this->create_mock_instance();
+
+		$instance->method( 'get_meta_value' )
+	         ->will(
+		         $this->returnValueMap(
+			         array(
+				         array( 1, 'public', '0' ),
+				         array( 1, 'latitude', self::MOCK_LAT ),
+				         array( 1, 'longitude', self::MOCK_LONG ),
+				         array( 1, 'address', self::MOCK_ADDRESS )
+			         )
+		         )
+	         );
+
+		return $instance;
+	}
+
+	/**
+	 * @param string[] $additional_mock_methods
+	 * @param boolean $disable_constructor
+	 * @return Jetpack_Geo_Location|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function create_mock_instance(
+		$additional_mock_methods = array(),
+		$disable_constructor = self::DISABLE_CONSTRUCTOR
+	) {
+		$mock_methods = array_merge(
+			array( 'get_meta_value', 'is_single', 'is_feed' ),
+			$additional_mock_methods
+		);
+
+		/* @var $instance Jetpack_Geo_Location|PHPUnit_Framework_MockObject_MockObject */
+		$builder = $this->getMockBuilder( Jetpack_Geo_Location::class )
+			->setMethods( $mock_methods );
+
+		if ( $disable_constructor ) {
+			$builder->disableOriginalConstructor();
+		}
+
+        return $builder->getMock();
```

Comment:
Missing tab? :)

Reply:
Did a quick pass with CodeSniffer on little style issues like this in 39df783.